diff --git a/dist/src/internal/builtin-plugins/solidity/build-system/resolver/npm-module-parsing.js b/dist/src/internal/builtin-plugins/solidity/build-system/resolver/npm-module-parsing.js
index f7a9380d3b617684a9f33957930e3dbc5d6a02f0..762b5e5ae7f8858fd2094ed2ffb8f4d90393059c 100644
--- a/dist/src/internal/builtin-plugins/solidity/build-system/resolver/npm-module-parsing.js
+++ b/dist/src/internal/builtin-plugins/solidity/build-system/resolver/npm-module-parsing.js
@@ -1,6 +1,6 @@
 import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
-const NPM_MODULE_PATTERN = /^(?<package>(?:@[a-z0-9]+(?:[._-][a-z0-9]+)*\/)?[a-z0-9]+(?:[._-][a-z0-9]+)*)\/(?<path>.+)$/;
-const NPM_MODULE_OR_PREFIX_PATTERN = /^(?<package>(?:@[a-z0-9]+(?:[._-][a-z0-9]+)*\/)?[a-z0-9]+(?:[._-][a-z0-9]+)*)\/?.*$/;
+const NPM_MODULE_PATTERN = /^(?<package>(?:@[a-z0-9]+(?:[._-][a-z0-9]+)*\/)?[a-z0-9]+(?:[._-][a-z0-9]+)*)\/(?<path>.+)$/i;
+const NPM_MODULE_OR_PREFIX_PATTERN = /^(?<package>(?:@[a-z0-9]+(?:[._-][a-z0-9]+)*\/)?[a-z0-9]+(?:[._-][a-z0-9]+)*)\/?.*$/i;
 /**
  * Parses a direct import as if it were an npm import, returning `undefined`
  * if the format is invalid.
